### PR TITLE
Docs: Demo and code samples

### PR DESCRIPTION
**Pull request by joshq00**
Beginning stages of a demo page that reflects the [mdl](http://www.getmdl.io/) samples (almost) 1:1

There are minor differences due to the default props in react-mdl

Complete examples:
* [x] Badges
* [x] Buttons
* [x] Cards
* [x] Layout
* [x] Loading
* [x] Menus
* [x] Sliders
* [x] Toggles
* [x] Tables
* [x] Text Fields
* [x] Tooltips

demo

### CODE REVIEW

**Comment by faergeek** [comment]
I just leave here these two links to not forget about it:
https://www.npmjs.com/package/react-styleguide-generator
https://github.com/reactjs/react-docgen

**Comment by tleunen** [comment]
Nice work @joshq00. Just by cruriosity, have you seen the `docs` branch (https://github.com/tleunen/react-mdl/tree/docs).

Some time ago, I started working on a real website for the documentation/demos. It's not updated

**Comment by faergeek** [comment]
@joshq00 Do you need some help with that?

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,5 @@
+root = true
+
+[*]
+indent_style = space
+indent_size = 4
```
Comment (tleunen) [inline .editorconfig]: :+1: 
I wanted to add it for a long time now

**Comment by faergeek** [comment]
I still would like to make something like live demo here https://gaearon.github.io/react-hot-loader/

File: demo/demo.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Layout, { Drawer, Header, Navigation, Content } from '../lib/Layout';
```
Comment (tleunen) [inline demo/demo.js]: should be `../src/Layout` :)

**Comment by tleunen** [comment]
When you click on a section, table for example, and then decide to scroll back up (at the beginning), the whole section is blank.

When inspecting, it goes on the table section even if I scrolled up.
screen shot 2015-10-24 at 5 27 29 pm

**Comment by tleunen** [comment]
Thank you very much for the contribution @joshq00, but unfortunately I was already working on one site. Feel free to take a look and contribute to it if you want! :)
http://tleunen.github.io/react-mdl/

**Pull request closed without merging**